awsbestpractices/checking-object-presence/CheckingObjectPresence.java

```diff
@@ -0,0 +1,239 @@
+package example;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.HeadBucketRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+public class Example {
+    AmazonS3 s3;
+    S3ObjectMetadata objectMetadata;
+
+    // Can be replaced
+    public boolean doesObjectExist_replacable_1() {
+        try {
+            s3.getObjectMetadata();
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    // Can be replaced
+    // Based on https://github.com/HubSpot/SlimFast/blob/59f74c9bf2d063d7d1e345fa1079d81435c5dcef/slimfast-plugin/src/main/java/com/hubspot/maven/plugins/slimfast/DefaultFileUploader.java#L44
+    private boolean doesObjectExist_replacable_2(String bucket, String key) throws MojoFailureException {
+        try {
+            s3.getObjectMetadata(bucket, key);
+            return true;
+        } catch (SdkBaseException e) {
+            if (e instanceof AmazonServiceException && ((AmazonServiceException) e).getStatusCode() == 404) {
+                return false;
+            } else {
+                throw new MojoFailureException("Error getting object metadata for key " + key, e);
+            }
+        }
+    }
+
+    // Can be replaced
+    // Based on https://code.amazon.com/packages/AWSSeerPlanSupport/blobs/ee7d0c39077b85bccdd94afed117a27ccfa12183/--/src/com/amazonaws/seer/spark/S3Util.java#L107-L121
+    private boolean doesObjectExist_replacable_3(AWSCredentialsProvider credentials, String path) {
+        AmazonS3URI s3BucketAndKey = new AmazonS3URI(path);
+        try {
+            AmazonS3 client = getClient(credentials);
+            String bucketName = s3BucketAndKey.getBucket();
+            client.getObject(bucketName, s3BucketAndKey.getKey());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is manually checking for the existence of an object. You could use the SDK to automatically check for this instead. Use the [doesObjectExist](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#doesObjectExist-java.lang.String-java.lang.String-) SDK operation.